botocore/data/emr/2009-03-31/waiters-2.json

```diff
@@ -37,6 +37,25 @@
           "expected": "TERMINATED_WITH_ERRORS"
         }
       ]
+    },
+    "ClusterTerminated": {
+      "delay": 30,
+      "operation": "DescribeCluster",
+      "maxAttempts": 60,
+      "acceptors": [
+        {
+          "state": "success",
+          "matcher": "path",
+          "argument": "Cluster.Status.State",
+          "expected": "TERMINATED"
+        },
+        {
+          "state": "failure",
```

Comment:
> So should we be raising an error if we reached the TERMINATED_WITH_ERRORS as the cluster did terminate? 

Yes I think we should be (i.e I agree with how it's currently written).

The most common use case here is to call `terminate` and then wait until the cluster is shutdown cleanly.  I think the fact that the service differentiates between `TERMINATED` and `TERMINATED_WITH_ERRORS` suggests we should not be lumping them together in our waiter.

I can't think of a reasonable use case where someone would want to terminate the cluster and wait for it to be terminated where they *wouldn't* want to differentiate the two states.  If they just wanted to terminate a cluster and didn't care about success/failure, I would expect them to not wait on it at all.

Additionally, a user can already get this behavior if they *really* wanted this:

```
try:
   wait()...
except WaiterError:
  pass
```